### PR TITLE
docs: install in non-package node

### PR DESCRIPTION
**Pull request by tchaikov**
when running `make setup`, we could have following failure:
```
Installing the current project: scylla (4.3.0)
The current project could not be installed: No file/folder found for package scylla
If you do not want to install the current project use --no-root
```

because docs is not a proper python project named "scylla",
and do not have a directory structure expected by poetry. what we
expect from poetry, is to manage the dependencies for building
the document.

so, in this change, we install in the `non-package` mode when running
`poetry install`, this skips the root package, which does not exist.
as an alternative, we could put an empty `scylla.py` under `docs`
directory, but that'd be overkill. or we could pass `--no-root`
to `poetry install`, but would be ideal if we can keep the settings
in a single place.

see also https://python-poetry.org/docs/basic-usage/#operating-modes,
and https://python-poetry.org/docs/cli/#options-2, for more
details on the settings and command line options of poetry.

please note this setting was added to poetry 1.8, so the required
poetry version is updated. we might need to upgrade poetry in existing
installation.

---

this change addresses the build failure when preparing the building environment of document, so no need to backport.

### CODE REVIEW

**Comment by tchaikov** [comment]
@dgarcia360 hi David, could you help review this change?

File: docs/Makefile
Comment (dgarcia360) [inline docs/Makefile:34]: Thanks @tchaikov! If possible, I'd prefer to add `package-mode = false` to the `pyproject.toml` file so that all Poetry-related configurations remain in the same file. Would you agree?

Comment (tchaikov) [reply docs/Makefile]: i agree. i blindly followed the suggestion. and i was using poetry 1.7.1. so didn't look into other alternatives.

**Comment by tchaikov** [comment]
v2:

- specify `non-package` mode in `tool.poetry` section of `pyproject.toml` instead of passing `--no-root` in command line
- bump up `poetry` to `>= 1.8`, as `non-package` was introduced in this version.

**Comment by tchaikov** [comment]
@dgarcia360 hi David, thank you for your review and suggestion. could you take another look?

**Comment by Annamikhlin** [comment]
> * bump up `poetry` to `>= 1.8`, as `non-package` was introduced in this version.

This change failing the docs-CI
[jenkins.scylladb.com/job/releng/job/scylladb-docs-CI/549/](https://jenkins.scylladb.com/job/releng/job/scylladb-docs-CI/549/)
For now all our local servers that runs docs-CI installed with `poetry==1.7.0`

**Comment by tchaikov** [comment]
> > * bump up `poetry` to `>= 1.8`, as `non-package` was introduced in this version.
> 
> This change failing the docs-CI [jenkins.scylladb.com/job/releng/job/scylladb-docs-CI/549/](https://jenkins.scylladb.com/job/releng/job/scylladb-docs-CI/549/) For now all our local servers that runs docs-CI installed with `poetry==1.7.0` @tchaikov - if `>= 1.8` is required. We need to update all our servers to version 1.8.3 (latest)?

@Annamikhlin i think so, on one hand, with `1.7.0`, `make setup` fails, and David suggests that we should use `non-package` option which was introduced in `1.8.0`. but on the other hand, to bump up to 1.8, out jenkins slaves should be updated. do we have a setting / jenkins playbook so we can update them?

**Comment by Annamikhlin** [comment]
> > > * bump up `poetry` to `>= 1.8`, as `non-package` was introduced in this version.
> > 
> > 
> > This change failing the docs-CI [jenkins.scylladb.com/job/releng/job/scylladb-docs-CI/549/](https://jenkins.scylladb.com/job/releng/job/scylladb-docs-CI/549/) For now all our local servers that runs docs-CI installed with `poetry==1.7.0` @tchaikov - if `>= 1.8` is required. We need to update all our servers to version 1.8.3 (latest)?
> 
> @Annamikhlin i think so, on one hand, with `1.7.0`, `make setup` fails, and David suggests that we should use `non-package` option which was introduced in `1.8.0`. but on the other hand, to bump up to 1.8, out jenkins slaves should be updated. do we have a setting / jenkins playbook so we can update them?

Yes, we have ansible playbook for this:
https://github.com/scylladb/scylla-pkg/blob/cf33b772ab556cb49343006f13ec228c65635d0d/ansible/roles/jenkins-node/tasks/fedora-setup.yml#L55

**Comment by tchaikov** [comment]
@Annamikhlin thanks. created https://github.com/scylladb/scylla-pkg/pull/4181

**Comment by Annamikhlin** [comment]
> @Annamikhlin thanks. created [scylladb/scylla-pkg#4181](https://github.com/scylladb/scylla-pkg/pull/4181)

Thanks! verification passed on one node that was upgraded https://jenkins.scylladb.com/job/releng/job/scylladb-docs-CI/551/

**Review by dgarcia360 — APPROVED** [review]
LGTM, as long as the docs pipeline is updated to use Poetry>=1.8.

**Comment by tchaikov** [comment]
thanks to @Annamikhlin all jenkins nodes updated.

@scylladb/scylla-maint hello maintainers, could you help queue this change?

**Pull request closed without merging**
